balancer-js/src/modules/pools/factory/composable-stable/composable-stable.factory.ts

```diff
@@ -0,0 +1,129 @@
+   *  * @param owner - The address of the owner of the pool
+   *  * @param amplificationParameter The amplification parameter(must be greater than 1)
+   *  * @param exemptFromYieldProtocolFeeFlags array containing boolean for each token exemption from yield protocol fee flags
+   *  * @returns a TransactionRequest object, which can be directly inserted in the transaction to create a composable stable pool
+   */
+  create({
+    contractAddress,
+    name,
+    symbol,
+    tokenAddresses,
+    amplificationParameter,
+    rateProviders,
+    tokenRateCacheDurations,
+    exemptFromYieldProtocolFeeFlags,
+    swapFee,
+    owner,
+  }: CreatePoolParameters): TransactionRequest {
+    const swapFeeScaled = scale(new BigNumber(swapFee), 18);
+
+    const params = [
+      name,
+      symbol,
+      tokenAddresses,
+      amplificationParameter,
+      rateProviders,
+      tokenRateCacheDurations,
+      exemptFromYieldProtocolFeeFlags,
+      swapFeeScaled.toString(),
+      owner,
+    ];
+    const composablePoolInterface = new Interface(composableStableAbi);
+    const createFunctionAbi = composableStableAbi.find(
+      ({ name }) => name === 'create'
+    );
+    if (!createFunctionAbi)
+      throw new BalancerError(BalancerErrorCode.INTERNAL_ERROR_INVALID_ABI);
+    const createFunctionFragment = FunctionFragment.from(createFunctionAbi);
+    const encodedFunctionData = composablePoolInterface.encodeFunctionData(
+      createFunctionFragment,
+      params
+    );
+    return {
+      to: contractAddress,
+      data: encodedFunctionData,
+    };
+  }
+
+  /**
+   * Build Init join pool transaction parameters (Can only be made once per pool)
+   * @param {JoinPoolParameters} params - parameters used to build exact tokens in for bpt out transaction
+   * @param {string}                          params.joiner - Account address joining pool
+   * @param {SubgraphPoolBase}                params.pool - Subgraph pool object of pool being joined
+   * @param {string[]}                        params.tokensIn - Token addresses provided for joining pool (same length and order as amountsIn)
+   * @param {string[]}                        params.amountsIn -  - Token amounts provided for joining pool in EVM amounts
+   * @param {string}                          wrappedNativeAsset - Address of wrapped native asset for specific network config. Required for joining with ETH.
+   * @returns                                 transaction request ready to send with signer.sendTransaction
+   */
+  buildInitJoin({
+    joiner,
+    poolId,
+    poolAddress,
+    tokensIn,
+    amountsIn,
+    wrappedNativeAsset,
+  }: InitJoinPoolParameters): InitJoinPoolAttributes {
+    const assetHelpers = new AssetHelpers(wrappedNativeAsset);
+    // sort inputs
+    const [sortedTokens, sortedAmounts] = assetHelpers.sortTokens(
+      tokensIn,
+      amountsIn
+    ) as [string[], string[]];
+    const userData = ComposableStablePoolEncoder.joinInit(sortedAmounts);
```

Comment:
For ComposableStable pools, userData should not contain the BPT amount, so you'll have to filter it out.
Here is a code snippet I got from generalisedJoin that should work for you:
```
    // userData amounts should not include the BPT of the pool being joined
    let userDataAmounts = [];
    const bptIndex = sortedTokens
      .map((t) => t.toLowerCase())
      .indexOf(node.address.toLowerCase());
    if (bptIndex === -1) {
      userDataAmounts = sortedAmounts;
    } else {
      userDataAmounts = [
        ...sortedAmounts.slice(0, bptIndex),
        ...sortedAmounts.slice(bptIndex + 1),
      ];
    }
    const userData = ComposableStablePoolEncoder.joinInit(userDataAmounts);
```